checkapi/internal/ast_test.go

```diff
@@ -126,6 +127,30 @@ func TestAST(t *testing.T) {
 			},
 			"chan(string)",
 		},
+		{
+			"generics",
+			&ast.FuncType{
+				TypeParams: &ast.FieldList{List: []*ast.Field{
+					{
+						Names: []*ast.Ident{
+							ast.NewIdent("T"),
+						},
+						Type: ast.NewIdent("T ~string"),
+					},
+				}},
+				Params: &ast.FieldList{List: []*ast.Field{
+					{
+						Names: []*ast.Ident{
+							ast.NewIdent("foo"),
+						},
+						Type: ast.NewIdent("T"),
+					},
+				},
+				},
+				Results: nil,
+			},
+			"func[T ~string](T) ",
```

Comment:
Is there meant to be a space on the end of the string?

Reply:
I probably could try harder and remove that

Reply:
https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/740 is open to follow up.